### ISSUE TITLE
Valiu Feedback

### ISSUE THREAD

**Issue by S0c5** [Sgarcia10/url-screenshot#1]
Hello Santiago,

Thanks for sending us your test, I'd like to ask you some questions related:

1. Why did you choose a sync architecture knowing the constraints of ~1K requests per second, seeing your code I saw that you raise a Web instance per each request - if I saw wrong please correct me - by sending ~1k requests at the same second your server gonna die. What other pattern allows you to accept ~ 1k per second without the need for high resources?

**Comment by Sgarcia10** [comment]
Yes you're right. Yesterday, when I finished the test, I realized how
expensive in performance was doing things synchronously because of webpage
screenshot processing. Rather, solution would be responding the s3 url
immediately and then start queuing screenshots tasks (sqs, Kafka, rabbitmq,
...)  and listen to them in a dedicated microservice (takes the screenshot
and stores into s3).
If you give a couple of days I'll show you the implementation.

I'll be waiting for your answer.

**Comment by S0c5** [comment]
Hello Santiago, I'd like to see your implementation in RabbitMq, Hoping for your contribution to review.

**Comment by Sgarcia10** [comment]
Hi David,
I apologize because  I have not enough time to set up a rabbitMq
environment. Sincerely, I'm more comfortable with sqs or kafka. Instead, I
prefer to improve my last answer because I was mistaken. I used serverless
infrastructure in order to prevent server death. Lambda will call the
function every time an event occurs, in this case a http post. So, there's
no option to server dies. Pricing is the only reason to change architecture
and it depends on how many requests per month the server will have and the
time server takes for code to execute; in my recent load testing it is
about 1 second average per request.
I'm looking forward to your response.

Best,

Santiago Garcia S

**Comment by S0c5** [comment]
Hello Santiago,

Thanks for writing to me, lets do an implementation with SQS or Kafka to see how do you handle it, I like your idea with lambda, let me know if you are able to do that implementation.

Regards,
David

**Comment by Sgarcia10** [comment]
Hi David,
I'm using lambda in my current implementation, let me do a queue
implementation until friday.

Sincerely,

Santiago Garcia
